src/services/swap_service.ts

```diff
@@ -315,6 +326,69 @@ export class SwapService implements ISwapService {
+        let sellTokenFeeAmount = ZERO; // total sell token fee to charge
+        let sellTokenFeeAmounts: AffiliateFeeAmount[] = []; // sell token fee amounts used by affiliate fee transformer
+        let sellTokenFeeOnChainTransferGas = ZERO; // the gas cost for transferring sell token fee on-chain
+        if (metaTransactionVersion !== undefined && marketSide === MarketOperation.Sell) {
+            // Use sell token as fee
+
+            // The check is only intended for compile time type checking
+            if (!sellAmount) {
+                throw new Error('sellAmount is undefined when market direction is sell');
+            }
+
+            feeToken = sellToken;
+            if (!providedGasPrice) {
+                providedGasPrice = await this._swapQuoter.getGasPriceEstimationOrThrowAsync();
+            }
+
+            // Only get `sellTokenAmountPerWei` when necessary
+            const sellTokenAmountPerWei = feeConfigs?.gasFee
+                ? await this._swapQuoter.getTokenAmountPerWei(sellToken, {})
+                : undefined;
+            // Need to calculate all fees in order to adjust `sellAmount` before passing it down to sampler & router
+            ({
+                sellTokenFees,
+                sellTokenFeeAmount,
+                sellTokenFeeOnChainTransfers: sellTokenFeeAmounts,
+                sellTokenFeeOnChainTransferGas,
+            } = this._getSellTokenFees(
+                feeConfigs,
+                sellToken,
+                sellAmount,
+                sellTokenAmountPerWei,
+                providedGasPrice,
+                AVG_MULTIPLEX_TRANFORM_ERC_20_GAS, // use historic average gas cost for multiplex & transformERC20
+                metaTransactionVersion === 'v1' ? TRANSFER_GAS : TRANSFER_FROM_GAS, // meta-transaction v1 uses affiliate fee transformer which calls `transformerTransfer` for on-chain fee transfer.
+                // meta-transaction v2 uses `FixinTokenSpender._transferERC20TokensFrom`.
+            ));
+
+            if (sellAmount.lte(sellTokenFeeAmount)) {
+                logger.info(
+                    {
+                        sellTokenFees,
+                        sellTokenFeeAmount,
+                        sellTokenFeeAmounts,
+                        sellTokenFeeOnChainTransferGas,
+                        sellAmount,
+                    },
+                    'sellAmount <= sell token fee amount',
+                );
+                throw new InsufficientFundsError(`sellAmount too small`);
+            }
+        }
+
+        const amount =
+            marketSide === MarketOperation.Sell
+                ? sellAmount?.minus(sellTokenFeeAmount)
+                : // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- TODO: fix me!
+                  buyAmount!.times(getBuyTokenPercentageFeeOrZero(affiliateFee) + 1).integerValue(BigNumber.ROUND_DOWN);
```

Comment:
this line seems fairly cryptic - I don't understand it without a comment... I guess it's not something you actually changed, but you will now get the git blame 😆

Reply:
Yeah let me move it back to where it was and added comment. I only changed the line on `sellAmount` and did not touch the line related to `buyAmount`